src/bun.js/bindings/JSReadableState.cpp

```diff
@@ -201,38 +146,133 @@ JSC_DEFINE_CUSTOM_GETTER(jsReadableState_pipesCount, (JSGlobalObject * lexicalGl
-}
 
-JSC_DEFINE_CUSTOM_SETTER(setJSReadableState_paused, (JSGlobalObject * lexicalGlobalObject, EncodedJSValue thisValue, EncodedJSValue encodedValue, PropertyName attributeName))
-{
-    auto& vm = JSC::getVM(lexicalGlobalObject);
-    auto throwScope = DECLARE_THROW_SCOPE(vm);
-    JSReadableState* state = JSC::jsDynamicCast<JSReadableState*>(JSValue::decode(thisValue));
-    if (!state) {
-        RETURN_IF_EXCEPTION(throwScope, false);
+JSReadableState_NULLABLE_BOOLEAN_GETTER_SETTER(paused)
+JSReadableState_NULLABLE_BOOLEAN_GETTER_SETTER(flowing)
+
+#undef JSReadableState_NULLABLE_BOOLEAN_GETTER_SETTER
+
+#define JSReadableState_GETTER_SETTER(NAME, TYPE)                                                                                      \
+    static JSC_DECLARE_CUSTOM_GETTER(jsReadableState_##NAME); \
+    JSC_DEFINE_CUSTOM_GETTER(jsReadableState_##NAME, (JSGlobalObject * lexicalGlobalObject, EncodedJSValue thisValue, PropertyName attributeName)) \
+    {                                                                                                                                              \
+        auto& vm = JSC::getVM(lexicalGlobalObject); \
+        auto throwScope = DECLARE_THROW_SCOPE(vm); \
+        JSReadableState* state = JSC::jsDynamicCast<JSReadableState*>(JSValue::decode(thisValue)); \
+        if (!state) { \
+            RETURN_IF_EXCEPTION(throwScope, JSC::JSValue::encode(JSC::jsUndefined())); \
+        } \
+        RELEASE_AND_RETURN(throwScope, JSC::JSValue::encode(JSC::js##TYPE(state->m_##NAME))); \
+    } \
+    \
+    static JSC_DECLARE_CUSTOM_SETTER(setJSReadableState_##NAME); \
+    JSC_DEFINE_CUSTOM_SETTER(setJSReadableState_##NAME, (JSGlobalObject * lexicalGlobalObject, EncodedJSValue thisValue, EncodedJSValue encodedValue, PropertyName attributeName)) \
+    { \
+        auto& vm = JSC::getVM(lexicalGlobalObject); \
+        auto throwScope = DECLARE_THROW_SCOPE(vm); \
+        JSReadableState* state = JSC::jsDynamicCast<JSReadableState*>(JSValue::decode(thisValue)); \
+        if (!state) { \
+            RETURN_IF_EXCEPTION(throwScope, false); \
+        } \
+        state->m_##NAME = JSC::JSValue::decode(encodedValue).to##TYPE(lexicalGlobalObject); \
+        RETURN_IF_EXCEPTION(throwScope, false); \
+        RELEASE_AND_RETURN(throwScope, true); \
     }
-    state->m_paused = JSC::JSValue::decode(encodedValue).toBoolean(lexicalGlobalObject) ? 1 : -1;
-    RELEASE_AND_RETURN(throwScope, true);
-}
+
+#define JSReadableState_BOOLEAN_GETTER_SETTER(NAME)       \
+    JSReadableState_GETTER_SETTER(NAME, Boolean)
+
+#define JSReadableState_NUMBER_GETTER_SETTER(NAME)        \
+    JSReadableState_GETTER_SETTER(NAME, Number)
+
+JSReadableState_BOOLEAN_GETTER_SETTER(ended)
+JSReadableState_BOOLEAN_GETTER_SETTER(endEmitted)
+JSReadableState_BOOLEAN_GETTER_SETTER(reading)
+JSReadableState_BOOLEAN_GETTER_SETTER(constructed)
```

Comment:
For getters/setters, I think we can assume that the type is always `JSReadableState`.